core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -102,6 +119,95 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
       case ChartType.SPLINE:
         this.generateChart();
         break;
+      case ChartType.SPATIAL_SCATTERPLOT:
+        this.generateSpatialScatterplot();
+        break;
+      case ChartType.SCATTERPLOT:
+        this.generateSimpleScatterplot();
+        break;
+    }
+  }
+
+  generateSimpleScatterplot() {
+    if (this.c3ChartElement) {
+      this.c3ChartElement.destroy();
+    }
+    const result = this.data as Array<Record<string, Primitive>>;
+    const xLabel: string = Object.keys(result[0])[0];
+    const yLabel: string = Object.keys(result[0])[1];
+
+    this.c3ChartElement = c3.generate({
+      size: {
+        height: VisualizationPanelContentComponent.HEIGHT,
+        width: VisualizationPanelContentComponent.WIDTH
+      },
+      data: {
+        json: result,
+        keys: {
+          x: xLabel,
+          value: [yLabel]
+        },
+        type: this.chartType as c3.ChartType
+      },
+      axis: {
+        x: {
+          label: xLabel,
+          tick: {
+            fit: true
+          }
+        },
+        y: {
+          label: yLabel
+        }
+      },
+      bindto: VisualizationPanelContentComponent.CHART_ID
+    });
+  }
+
+  generateSpatialScatterplot() {
+    if (this.map === undefined) {
+      this.initMap();
+    }
+    this.map?.on('styledata', () => {
+      this.addScatterLayer();
+    });
```

Comment:
I am not very familiar with the library here. can you add some comments to explain when will this `styledata` event be triggered/fired? I did some search and the condition of this event is not clear to me.

Reply:
once the map tile(base) is rendered, then we can add the layer of points otherwise, we would get an error. this will be handled in the new PR as I will render the map in the background

Reply:
When would this event be triggered?

Reply:
I was reading this post and wonder if we are using it incorrectly. https://stackoverflow.com/questions/58223541/mapbox-styledata-event-fire-3-times-i-only-need-fire-1-time-how. Can you please verify it? Also it will be good to leave a comment about the trigger condition in code.

Reply:
this will be replaced in a future PR anyways. This PR is temporary to change only a few files and lines of code to make the review easier